### ISSUE TITLE
pl-order-blocks without a mouse (keyboard accessibility)

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#4912]
When answering on a touch only device (e.g., phone) the pl-order-blocks element doesn't provide a way to provide an answer. I imagine a similar issue could also affect accessibility. I'm wondering if something like small buttons to add entries to the answer and to reorder them may address both of these issues.

**Comment by SethPoulsen** [comment]
I have a fix for touch screens here: https://github.com/PrairieLearn/PrairieLearn/pull/4954

This issue should remain open because Accessibility for people with low vision who navigate only with a keyboard is still a problem and it is something that I would like to support when I get around to it.

**Comment by nwalters512** [comment]
A possible starting point: https://www.smashingmagazine.com/2018/01/dragon-drop-accessible-list-reordering/

There's some good discussion happening on [this Slack thread](https://prairielearn.slack.com/archives/C0XBTPZMZ/p1634661735378500) around this; lifting a few thoughts into this issue for when free Slack inevitably wipes those messages from history 😢:

* We could probably use left/right and MOD+left/MOD+right to control indentation and moving items between the choices and answer zones, where MOD is shift or command or some other modifier. It's not clear which keys would be better suited for indentation vs. moving between dropzones.
* We probably want focus to stay with items as they're moved around.

**Comment by SethPoulsen** [comment]
It seems some other people have hacked accessibility features onto the jqueryui sortable, which is what we're using: 

https://codepen.io/bartz/pen/rrgyjw

https://stackoverflow.com/questions/16844616/jqueryui-sortable-with-keyboard

**Comment by echuber2** [comment]
If touch has been resolved in https://github.com/PrairieLearn/PrairieLearn/pull/4954 then maybe we should rename this issue to focus on accessibility. (I couldn't find it until Seth gave me the link.)

**Comment by echuber2** [comment]
Discord has an MIT-licensed library for accessible drag-and-drop (if I'm reading this correctly): https://github.com/discord/react-dnd-accessible-backend

**Comment by SethPoulsen** [comment]
@echuber2 Looks cool thanks for the reference! I obviously can't use that since it got a million dependencies, but good to know it exits. I also was unaware of the new HTML 'draggable' attribute & friends. I'll be sure to look into that when I get to making sure the elements are properly annotated for screen readers.
